snapfaas/bins/multivm/main.rs

```diff
@@ -85,57 +111,97 @@ fn main() {
             e => panic!("Cannot create \":{}\". {:?}", image, e),
         }
     }
-    // create the resource manager
-    let (mut manager, manager_sender) = ResourceManager::new(config);
+    // create the local resource manager
+    let (mut manager, manager_sender) = ResourceManager::new(config, sched_sa.clone());
 
     // set total memory
-    let total_mem = matches.value_of("total memory").unwrap()
-        .parse::<usize>().expect("Total memory is not a valid integer");
+    let total_mem = matches
+                        .value_of("total memory")
+                        .unwrap()
+                        .parse::<usize>()
+                        .expect("Total memory is not a valid integer");
     manager.set_total_mem(total_mem);
 
     // create the worker pool
-    let (pool, request_sender) = new_workerpool(manager.total_mem()/128, manager_sender.clone());
+    let pool = new_workerpool(manager.total_mem()/128, sched_sa.clone(), manager_sender.clone());
     // kick off the resource manager
     let manager_handle = manager.run();
 
     // register signal handler
-    set_ctrlc_handler(request_sender.clone(), pool, manager_sender, Some(manager_handle));
+    set_ctrlc_handler(pool, sched_sa.clone(), manager_sender, Some(manager_handle));
 
     // TCP gateway
-    if let Some(l) = matches.value_of("listen address") {
-        let gateway = gateway::HTTPGateway::listen(l);
-
-        for (request, response_tx, timestamps) in gateway {
+    if let Some(l) = matches.value_of("http listen address") {
+        let gateway = snapfaas::sched::gateway::HTTPGateway::listen(l, None);
+        for (request, tx, _timestamps) in gateway {
             // Return when a VM acquisition succeeds or fails
             // but before a VM launches (if it is newly allocated)
             // and execute the request.
-            request_sender.send(Message::Request((request, response_tx, timestamps))).expect("Failed to send request");
+            if let Some(_) = schedgate.next() {
+                let sched_resman_dup = Arc::clone(&sched_resman);
+                thread::spawn(move || {
+                    let _ = snapfaas::sched::schedule_sync(request, sched_resman_dup, tx);
+                });
+            }
         }
     }
-}
 
-fn new_workerpool(pool_size: usize, manager_sender: Sender<Message>) -> (Vec<Worker>, Sender<Message>) {
-    let (request_sender, response_receiver) = mpsc::channel();
-    let response_receiver = Arc::new(Mutex::new(response_receiver));
+    // // TCP gateway
```

Comment:
ditto, just remove commented code. it's there in the git history if we ever need it.